mixins/property-required/property-required-mixin.js

```diff
@@ -80,13 +71,18 @@ export const PropertyRequiredMixin = dedupeMixin(superclass => class extends sup
 		clearTimeout(info.timeout);
 		info.timeout = null;
 
-		const hasValue = this[name]?.constructor === String && this[name]?.length > 0;
-		const success = info.validator(hasValue);
+		if (info.type !== undefined && info.type !== String) {
```

Comment:
I moved the type checking here since otherwise there was no way to test it (custom elements constructors that throw errors can't be caught).